pkg/scheduler/plugin.go

```diff
@@ -615,13 +615,12 @@ func (k *deleter) deleteOne(pod *Pod) error {
 	}
 }
 
-// Create creates a scheduler plugin and all supporting background functions.
+// Create a scheduler plugin and all supporting background functions.
 func (k *KubernetesScheduler) NewPluginConfig(terminate <-chan struct{}, mux *http.ServeMux) *PluginConfig {
-
 	// Watch and queue pods that need scheduling.
 	updates := make(chan queue.Entry, k.schedcfg.UpdatesBacklog)
 	podUpdates := &podStoreAdapter{queue.NewHistorical(updates)}
-	reflector := cache.NewReflector(createAllPodsLW(k.client), &api.Pod{}, podUpdates, 0)
+	reflector := cache.NewReflector(k.podsListWatch, &api.Pod{}, podUpdates, 0)
```

Comment:
I bet! It's tedious sometimes to figure out exactly how to get at mocking
things.

On Wed, May 20, 2015 at 4:30 PM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In pkg/scheduler/plugin.go
> <https://github.com/mesosphere/kubernetes-mesos/pull/274#discussion_r30745644>
> :
>
> >  	// Watch and queue pods that need scheduling.
> >  	updates := make(chan queue.Entry, k.schedcfg.UpdatesBacklog)
> >  	podUpdates := &podStoreAdapter{queue.NewHistorical(updates)}
> > -	reflector := cache.NewReflector(createAllPodsLW(k.client), &api.Pod{}, podUpdates, 0)
> > +	reflector := cache.NewReflector(k.podsListWatch, &api.Pod{}, podUpdates, 0)
>
> Agree, that would be cleaner. Was happy to get that mocked without faking
> the whole apiserver watch API first.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/kubernetes-mesos/pull/274/files#r30745644>.
>